lib/disposable.dart

```diff
@@ -101,21 +141,34 @@ abstract class Disposable {
   @mustCallSuper
   @protected
   void manageDisposable(Disposable disposable) {
-    _disposables.add(disposable);
+    _internalDisposables.add(disposable);
+  }
+
+  /// Automatically handle arbitrary disposals using a callback.
+  @mustCallSuper
+  @protected
+  void manageDisposer(Disposer disposer) {
```

Comment:
I'm trying to find good use cases for the new func library in w_common. Maybe this is one? Rather than defining a `Disposer` typedef you could use `Func<Future<dynamic>>`. :shrug: